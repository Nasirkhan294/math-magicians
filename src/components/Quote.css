.quotes {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 50%;
}

.quote {
  font-size: 2rem;
  width: 100%;
}
