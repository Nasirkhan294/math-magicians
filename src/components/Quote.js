import { useEffect, useState } from 'react';
import './Quote.css';

function Quote() {
  const [quote, setQuote] = useState(null);
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState(null);

  useEffect(() => {
    fetch('https://api.api-ninjas.com/v1/quotes?category=knowledge', {
      headers:
        { 'X-Api-Key': 'A5gXbih2P+SXXKb1RLdvfw==7imgYlRKUhRp6KRD' },
    })
      .then((resp) => resp.json())
      .then((apiData) => {
        setLoading(false);
        const body = `${apiData[0].quote} - ${apiData[0].author}`;
        setQuote(body);
      })
      .catch((error) => {
        setLoading(false);
        setError(error);
      });
  }, []);

  if (loading) {
    return <div>Loading...</div>;
  }

  if (error) {
    return (
      <div>
        Error:
        {error.message}
      </div>
    );
  }

  return (
    <div className="quotes">
      <q className="quote">{quote}</q>
    </div>
  );
}

export default Quote;
