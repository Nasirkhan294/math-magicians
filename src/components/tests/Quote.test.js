import React from 'react';
import { render } from '@testing-library/react';
import Quote from '../Quote';

test('Quote matches snapshot', () => {
  const { asFragment } = render(<Quote />);
  expect(asFragment()).toMatchSnapshot();
});
