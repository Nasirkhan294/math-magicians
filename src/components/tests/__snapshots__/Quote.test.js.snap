// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote matches snapshot 1`] = `
<DocumentFragment>
  <div>
    Loading...
  </div>
</DocumentFragment>
`;
